### PR TITLE
Update tests for caliper delivery point

### PR DESCRIPTION
**Pull request by mumarkhan999**

### CODE REVIEW

File: openedx_caliper_tracking/tests/tests.py
Comment (danialmalik) [inline openedx_caliper_tracking/tests/tests.py]:
```suggestion
        If the settings ENABLE_CALIPER_EVENTS_DELIVERY, CALIPER_DELIVERY_ENDPOINT
```

File: openedx_caliper_tracking/tests/tests.py
Comment (danialmalik) [inline openedx_caliper_tracking/tests/tests.py:92]: Please use patch.dict for changing the dict values.
https://docs.python.org/3/library/unittest.mock.html#patch-dict

**Review by danialmalik — CHANGES_REQUESTED** [review]

File: openedx_caliper_tracking/tests/tests.py
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py]: done.

File: openedx_caliper_tracking/tests/tests.py
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:92]: done.

Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:92]: In this scenario, it is not possible but I have used it wherever it is possible.

File: openedx_caliper_tracking/tests/tests.py
Comment (tasawernawaz) [inline openedx_caliper_tracking/tests/tests.py:93]: why this? you should override settings.

**Review by imhassantariq — APPROVED** [review]

**Review by tehreem-sadat — APPROVED** [review]

File: openedx_caliper_tracking/tests/tests.py
Comment (tasawernawaz) [inline openedx_caliper_tracking/tests/tests.py]: can we write simple docstring? 
```
Test that API won't call when `ENABLE_CALIPER_EVENTS_DELIVERY` is not set.
```

File: openedx_caliper_tracking/tests/tests.py
Comment (tasawernawaz) [reply openedx_caliper_tracking/tests/tests.py:93]: usually, we start docstring in tests with `Test` or `Verify` for consistency and readability.

File: openedx_caliper_tracking/tests/tests.py
Comment (tasawernawaz) [inline openedx_caliper_tracking/tests/tests.py:144]: why change on this line ?

**Review by tasawernawaz — CHANGES_REQUESTED** [review]

File: openedx_caliper_tracking/tests/tests.py
Comment (danialmalik) [reply openedx_caliper_tracking/tests/tests.py:144]: trailing `,` is removed

File: openedx_caliper_tracking/tests/tests.py
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:93]: > why this? you should override settings.

Actually here I am deleting these settings if exists. Before these changes, our tests were failing as we were trying to delete the settings which weren't present in the settings object.

File: openedx_caliper_tracking/tests/tests.py
Comment (mumarkhan999) [reply openedx_caliper_tracking/tests/tests.py:144]: just for the consistency, I removed trailing commas

**Comment by mumarkhan999** [comment]
@tasawernawaz All the docstrings have been updated in the latest commit.

**Review by danialmalik — APPROVED** [review]

**Comment by mumarkhan999** [comment]
@tasawernawaz waiting for your approval/review.

**Review by tasawernawaz — APPROVED** [review]

**Pull request merged**
